### DISCUSSION TITLE
ALPR 3.3.4 - Docker on Ubuntu - Will not install GPU capabilities

### DISCUSSION THREAD

**Discussion by kevinruffus** [codeproject/CodeProject.AI-Server#237] [Q&A]
I haven't been able to get anything but Face Processing and YOLOv5 6.2 to run off the GPU for several releases. Again ALPR refuses to install GPU libraries and use the GPU.
Obvious bit from installation logs:

20:19:12:ALPR: GPU support                                                           
20:19:12:ALPR: CUDA (NVIDIA) Present: Yes (CUDA 12.2, cuDNN 8.9.6)
20:19:12:ALPR: ROCm (AMD) Present:    No
20:19:12:ALPR: MPS (Apple) Present:   No
20:19:13:ALPR: Reading module settings.......done
20:19:13:ALPR: Processing module ALPR 3.3.4 (Internal)                               
20:19:13:ALPR: Variable Dump                                                         
20:19:13:ALPR: moduleName          = License Plate Reader
20:19:13:ALPR: moduleId            = ALPR
20:19:13:ALPR: moduleVersion       = 3.3.4
20:19:13:ALPR: runtime             = python3.8
20:19:13:ALPR: runtimeLocation     = Local
20:19:13:ALPR: installGPU          = false
20:19:13:ALPR: pythonVersion       = 3.8
20:19:13:ALPR: virtualEnvDirPath   = /app/modules/ALPR/bin/ubuntu/python38/venv
20:19:13:ALPR: venvPythonCmdPath   = /app/modules/ALPR/bin/ubuntu/python38/venv/bin/python3.8
20:19:13:ALPR: packagesDirPath     = /app/modules/ALPR/bin/ubuntu/python38/venv/lib/python3.8/site-packages/
20:19:13:ALPR: moduleStartFilePath = ALPR_adapter.py
20:19:13:ALPR: Installing Python 3.8
20:19:13:ALPR: Python install path is /app/modules/ALPR/bin/ubuntu/python38
20:19:13:ALPR: Python 3.8 is already installed
20:19:14:ALPR: Ensuring PIP in base python install...Reading package lists...
20:19:15:ALPR: Building dependency tree...
20:19:15:ALPR: Reading state information...
20:19:16:ALPR: 0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 48 not upgraded.
20:19:16:ALPR: Need to get 193 kB of archives.
20:19:16:ALPR: After this operation, 0 B of additional disk space will be used.
20:19:16:ALPR: Get:1 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 python3.8-distutils all 3.8.20-1+jammy1 [193 kB]
20:19:16:ALPR: debconf: delaying package configuration, since apt-utils is not installed
20:19:16:ALPR: Fetched 193 kB in 1s (214 kB/s)
20:19:16:ALPR: (Reading database ... 
20:19:16:ALPR: (Reading database ... 5%
20:19:16:ALPR: (Reading database ... 10%
20:19:16:ALPR: (Reading database ... 15%
20:19:16:ALPR: (Reading database ... 20%
20:19:16:ALPR: (Reading database ... 25%
20:19:16:ALPR: (Reading database ... 30%
20:19:16:ALPR: (Reading database ... 35%
20:19:16:ALPR: (Reading database ... 40%
20:19:16:ALPR: (Reading database ... 45%
20:19:16:ALPR: (Reading database ... 50%
20:19:16:ALPR: (Reading database ... 55%
20:19:16:ALPR: (Reading database ... 60%
20:19:16:ALPR: (Reading database ... 65%
20:19:16:ALPR: (Reading database ... 70%
20:19:16:ALPR: (Reading database ... 75%
20:19:16:ALPR: (Reading database ... 80%
20:19:16:ALPR: (Reading database ... 85%
20:19:16:ALPR: (Reading database ... 90%
20:19:16:ALPR: (Reading database ... 95%
20:19:16:ALPR: (Reading database ... 100%
20:19:16:ALPR: (Reading database ... 31004 files and directories currently installed.)
20:19:16:ALPR: Preparing to unpack .../python3.8-distutils_3.8.20-1+jammy1_all.deb ...
20:19:16:ALPR: Unpacking python3.8-distutils (3.8.20-1+jammy1) over (3.8.20-1+jammy1) ...
20:19:16:ALPR: Setting up python3.8-distutils (3.8.20-1+jammy1) ...
20:19:18:ALPR: Looking in links: /tmp/tmprlczr59q
20:19:18:ALPR: Requirement already satisfied: setuptools in /usr/local/lib/python3.8/dist-packages (75.3.0)
20:19:18:ALPR: Requirement already satisfied: pip in /usr/local/lib/python3.8/dist-packages (24.3.1)
20:19:19:ALPR: WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
20:19:19:ALPR: done
20:19:20:ALPR: Upgrading PIP in base python install...Requirement already satisfied: pip in /usr/local/lib/python3.8/dist-packages (24.3.1)
20:19:20:ALPR: WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.
20:19:20:ALPR: done
20:19:20:ALPR: Installing Virtual Environment tools for Linux...
20:19:20:ALPR: Searching for installed dependencies:
20:19:25:ALPR:  -> python3-pip python3-setuptools python3.8-venv done
20:19:25:ALPR: All dependencies already installed.
20:19:25:ALPR: Creating Virtual Environment (Local)...Install path is /app/modules/ALPR/bin/ubuntu/python38
20:19:28:ALPR: done
20:19:28:ALPR: Checking for Python 3.8...(Found Python 3.8.20) All good
20:19:31:ALPR: Upgrading PIP in virtual environment... done
20:19:32:Response rec'd from Object Detection (YOLOv5 6.2) command 'detect' (...17d196) ['No objects found']  took 42ms
20:19:33:ALPR: Installing updated setuptools in venv... done
20:19:33:ALPR: Downloading ocr-en-pp_ocrv4-paddle.zip to /app/downloads/modules/assets/ALPR
20:19:33:ALPR: Extracting to paddleocr in this folder
20:19:33:ALPR: Downloading OCR models... already exists...Expanding...Archive:  ocr-en-pp_ocrv4-paddle.zip
20:19:33:ALPR:   inflating: paddleocr/ch_ppocr_mobile_v2.0_cls_infer/inference.pdiparams  
20:19:33:ALPR:   inflating: paddleocr/ch_ppocr_mobile_v2.0_cls_infer/inference.pdiparams.info  
20:19:33:ALPR:   inflating: paddleocr/ch_ppocr_mobile_v2.0_cls_infer/inference.pdmodel  
20:19:33:ALPR:   inflating: paddleocr/en_PP-OCRv3_det_infer/inference.pdiparams  
20:19:33:ALPR:   inflating: paddleocr/en_PP-OCRv3_det_infer/inference.pdiparams.info  
20:19:33:ALPR:   inflating: paddleocr/en_PP-OCRv3_det_infer/inference.pdmodel  
20:19:33:ALPR:    creating: paddleocr/en_PP-OCRv4_rec_infer/
20:19:33:ALPR:   inflating: paddleocr/en_PP-OCRv4_rec_infer/inference.pdiparams  
20:19:33:ALPR:   inflating: paddleocr/en_PP-OCRv4_rec_infer/inference.pdiparams.info  
20:19:33:ALPR:   inflating: paddleocr/en_PP-OCRv4_rec_infer/inference.pdmodel  
20:19:33:ALPR: done.
20:19:33:ALPR: Moving contents of ocr-en-pp_ocrv4-paddle.zip to paddleocr...done.
20:19:33:ALPR: Installing Python packages for License Plate Reader
20:19:33:ALPR: Installing GPU-enabled libraries: No
20:19:33:ALPR: Ensuring PIP is installed and up to date...
20:19:33:ALPR: Searching for installed dependencies:
20:19:35:ALPR:  -> python3-pip done
20:19:35:ALPR: All dependencies already installed.

Docker on Ubuntu Server 22.04, Nvidia GTX 1080 with patched drivers (535.183.01), CUDA 12.2, server ver 2.9.7

Anyone have any ideas on how to get ALPR to use the GPU?

**Comment by kevinruffus** [comment]
![Screenshot 2024-12-18 203343](https://github.com/user-attachments/assets/76a1b79f-e030-40ee-b81f-6b424441946d)

**Comment by ChrisMaunder** [comment]
Unfortunately we have not been able to get GPU support on linux working reliably so it's no longer a supported option.
